jdaviz/app.py

```diff
@@ -1020,13 +1027,10 @@ def get_subsets(self, subset_name=None, spectral_only=False,
                 else:
                     all_subsets[label] = subset_region
 
-        # can this be done at the top to avoid traversing all subsets if only
-        # one is requested?
-        all_subset_names = [subset.label for subset in dc.subset_groups]
-        if subset_name and subset_name in all_subset_names:
+        if subset_name and subset_name in all_subsets:
             return all_subsets[subset_name]
         elif subset_name:
-            raise ValueError(f"{subset_name} not in {all_subset_names}")
+            raise ValueError(f"{subset_name} not in {all_subsets.keys()}")
```

Comment:
Do we want to sort this to be easier on human eyes? I guess for now it does not matter, because numbering is also insertion ordered, but might make a difference when we allow renaming subset?

Reply:
I think it's reasonable to sort, will do.